lib/taskana-core/src/test/java/acceptance/jobs/TaskCleanupJobAccTest.java

```diff
@@ -118,14 +123,59 @@ void should_DeleteOldTaskCleanupJobs_When_InitializingSchedule() throws Exceptio
     assertThat(jobsToRun).doesNotContainAnyElementsOf(taskCleanupJobs);
   }
 
-  private Task createAndCompleteTask() throws Exception {
+  @WithAccessId(user = "admin")
+  @Test
+  void should_DeleteCompletedTaskWithParentBusinessNull_When_RunningCleanupJob() throws Exception {
+    String taskId1 = createAndInsertTask(null);
+    taskService.claim(taskId1);
+    taskService.completeTask(taskId1);
+    String taskId2 = createAndInsertTask(null);
+    taskService.claim(taskId2);
+
+    long taskCount = taskService.createTaskQuery().idIn(taskId1, taskId2).count();
+    assertThat(taskCount).isEqualTo(2);
+
+    taskanaEngine.getConfiguration().setTaskCleanupJobAllCompletedSameParentBusiness(true);
+    taskanaEngine.getConfiguration().setCleanupJobMinimumAge(Duration.ZERO);
+    TaskCleanupJob job = new TaskCleanupJob(taskanaEngine, null, null);
+    job.run();
+
+    List<TaskSummary> tasksAfterCleaning =
+        taskService.createTaskQuery().idIn(taskId1, taskId2).list();
+    assertThat(tasksAfterCleaning).hasSize(1);
+    assertThat(tasksAfterCleaning.get(0).getId()).isEqualTo(taskId2);
+  }
+
+  @WithAccessId(user = "admin")
+  @Test
+  void should_DeleteCompletedTaskWithParentBusinessEmpty_When_RunningCleanupJob() throws Exception {
+    String taskId1 = createAndInsertTask("");
+    taskService.claim(taskId1);
+    taskService.completeTask(taskId1);
+    String taskId2 = createAndInsertTask("");
+    taskService.claim(taskId2);
+
+    long taskCount = taskService.createTaskQuery().idIn(taskId1, taskId2).count();
+    assertThat(taskCount).isEqualTo(2);
+
+    taskanaEngine.getConfiguration().setTaskCleanupJobAllCompletedSameParentBusiness(true);
+    taskanaEngine.getConfiguration().setCleanupJobMinimumAge(Duration.ZERO);
+    TaskCleanupJob job = new TaskCleanupJob(taskanaEngine, null, null);
+    job.run();
+
+    List<TaskSummary> tasksAfterCleaning =
+        taskService.createTaskQuery().idIn(taskId1, taskId2).list();
+    assertThat(tasksAfterCleaning).hasSize(1);
+    assertThat(tasksAfterCleaning.get(0).getId()).isEqualTo(taskId2);
+  }
```

Comment:
This test is a duplicate to the test above. Please use dynamic tests to increase maintainability.